packages/records/lib/models/records.ts

```diff
@@ -289,7 +289,7 @@ export async function update({
 
                     const { record: oldRecord, ...oldRecordRest } = rawOldRecord;
 
-                    const record = records.find((record) => record.external_id === oldRecord.id);
+                    const record = records.find((record) => record.external_id === rawOldRecord.external_id);
```

Comment:
I wonder why we were comparing the id with the external_id in the first place. That appears to be plain wrong which means this method didn’t work as intended to begin with.

Reply:
it was more of a naming issue. `record` is is the entire record we store in db. what was called `oldRecord` is actually the record data as defined in the model.